### PR TITLE
fix(Message): handle fetched messages

### PR DESCRIPTION
**Pull request by almostSouji**
**Please describe the changes this PR makes and why it should be merged:**

PR #6271 has introduced a regression, because fetched messages do not include a `guild_id` property in the rest response. This causes `Message#guild` to become `null` for fetched messages, even if they were fetched from a guild channel.

- try to assign the guild id from the channel, if available
- adapt the `Message#guild` getter to also return the guild from the `Message#channel` getter, if necessary and available

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

File: src/structures/Message.js [hunk 1]
```diff
@@ -390,7 +390,7 @@ class Message extends Base {
    * @readonly
    */
   get guild() {
-    return this.client.guilds.resolve(this.guildId);
+    return this.client.guilds.resolve(this.guildId) ?? this.channel?.guild ?? null;
```
Comment (kyranet) [inline src/structures/Message.js:393]: Is this change necessary with the constructor's assignment?

Comment (ckohen) [reply src/structures/Message.js:393]: Likelihood? No
Theres an extremely small niche use where you fetch a message, the channel is not cached, now you have no guiildId, channel gets cached, now you have guild

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]
